### PR TITLE
[Redis] Adding Managed Identity support

### PR DESCRIPTION
**Pull request by gorandom**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by gorandom** [comment]
@dingmeng-xue @msJinLei Can you check and merge this PR?

**Comment by gorandom** [comment]
@dingmeng-xue @msJinLei Can you check this PR?

File: src/RedisCache/RedisCache/Commands/NewAzureRedisCache.cs [hunk 1]
```diff
@@ -83,6 +83,12 @@ public class NewAzureRedisCache : RedisCacheCmdletBase
         [Parameter(ValueFromPipelineByPropertyName = true, Mandatory = false, HelpMessage = "Redis version. Valid values: 4, 6")]
         public string RedisVersion { get; set; }
 
+        [Parameter(ValueFromPipelineByPropertyName = true, Mandatory = false, HelpMessage = "Specifies the type of identity used for the Azure Cache for Redis. Valid values: \"SystemAssigned\" or \"UserAssigned\" or \"SystemAssignedUserAssigned\" or \"None\" ")]
+        public string IdentityType { get; set; }
```
Comment (msJinLei) [inline src/RedisCache/RedisCache/Commands/NewAzureRedisCache.cs:87]: Please provide a argument completer for IdentityType

File: src/RedisCache/RedisCache/Commands/SetAzureRedisCache.cs [hunk 2]
```diff
@@ -68,6 +68,12 @@ public class SetAzureRedisCache : RedisCacheCmdletBase
         [Parameter(ValueFromPipelineByPropertyName = true, Mandatory = false, HelpMessage = "A hash table which represents tags.")]
         public Hashtable Tag { get; set; }
 
+        [Parameter(ValueFromPipelineByPropertyName = true, Mandatory = false, HelpMessage = "Specifies the type of identity used for the Azure Cache for Redis. Valid values: \"SystemAssigned\" or \"UserAssigned\" or \"SystemAssignedUserAssigned\" or \"None\" ")]
+        public string IdentityType { get; set; }
```
Comment (msJinLei) [inline src/RedisCache/RedisCache/Commands/SetAzureRedisCache.cs:72]: Please provide a argument completer for IdentityType

**Review by msJinLei — CHANGES_REQUESTED** [review]

**Comment by msJinLei** [comment]
> @dingmeng-xue @msJinLei Can you check this PR?

Please refer to the comments, thanks

**Comment by msJinLei** [comment]
Related cmdlet reivew https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1100

File: src/RedisCache/RedisCache/Commands/SetAzureRedisCache.cs (hunk 2 above)
Comment (gorandom) [reply src/RedisCache/RedisCache/Commands/SetAzureRedisCache.cs:73]: added

File: src/RedisCache/RedisCache/Commands/NewAzureRedisCache.cs (hunk 1 above)
Comment (gorandom) [reply src/RedisCache/RedisCache/Commands/NewAzureRedisCache.cs:88]: added

File: src/RedisCache/RedisCache/Commands/SetAzureRedisCache.cs [hunk 3]
```diff
@@ -68,6 +68,13 @@ public class SetAzureRedisCache : RedisCacheCmdletBase
         [Parameter(ValueFromPipelineByPropertyName = true, Mandatory = false, HelpMessage = "A hash table which represents tags.")]
         public Hashtable Tag { get; set; }
 
+        [Parameter(ValueFromPipelineByPropertyName = true, Mandatory = false, HelpMessage = "Specifies the type of identity used for the Azure Cache for Redis. Valid values: \"SystemAssigned\" or \"UserAssigned\" or \"SystemAssignedUserAssigned\" or \"None\" ")]
+        [PSArgumentCompleter("SystemAssigned", "UserAssigned", "SystemAssignedUserAssigned", "None")]
+        public string IdentityType { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = true, Mandatory = false, HelpMessage = "Specifies one or more comma seperated user identities to be associated with the Azure Cache for Redis. The user identity references will be ARM resource ids in the form: '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/identities/{identityName}'")]
+        public string UserAssignedIdentity { get; set; }
+
```
Comment (msJinLei) [inline src/RedisCache/RedisCache/Commands/SetAzureRedisCache.cs:77]: Please also change string to string [] if you want to support  -UserAssignedIdentity "Id0", "Id1" as the input

File: src/RedisCache/RedisCache/Commands/NewAzureRedisCache.cs [hunk 4]
```diff
@@ -83,6 +83,13 @@ public class NewAzureRedisCache : RedisCacheCmdletBase
         [Parameter(ValueFromPipelineByPropertyName = true, Mandatory = false, HelpMessage = "Redis version. Valid values: 4, 6")]
         public string RedisVersion { get; set; }
 
+        [Parameter(ValueFromPipelineByPropertyName = true, Mandatory = false, HelpMessage = "Specifies the type of identity used for the Azure Cache for Redis. Valid values: \"SystemAssigned\" or \"UserAssigned\" or \"SystemAssignedUserAssigned\" or \"None\" ")]
+        [PSArgumentCompleter("SystemAssigned", "UserAssigned", "SystemAssignedUserAssigned", "None")]
+        public string IdentityType { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = true, Mandatory = false, HelpMessage = "Specifies one or more comma seperated user identities to be associated with the Azure Cache for Redis. The user identity references will be ARM resource ids in the form: '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/identities/{identityName}'")]
```
Comment (msJinLei) [inline src/RedisCache/RedisCache/Commands/NewAzureRedisCache.cs:90]: Please also change string to string [] if you want to support  -UserAssignedIdentity "Id0", "Id1" as the input

Comment (gorandom) [reply src/RedisCache/RedisCache/Commands/NewAzureRedisCache.cs:90]: modified

File: src/RedisCache/RedisCache/Commands/SetAzureRedisCache.cs (hunk 3 above)
Comment (gorandom) [reply src/RedisCache/RedisCache/Commands/SetAzureRedisCache.cs:77]: modified

**Review by msJinLei — APPROVED** [review]

**Pull request merged**